src/components/Modal/types.ts

```diff
@@ -0,0 +1,65 @@
+import {ValueOf} from 'type-fest';
+import {ViewStyle} from 'react-native';
+import {ModalProps} from 'react-native-modal';
+import {WindowDimensionsProps} from '../withWindowDimensions/types';
+import CONST from '../../CONST';
+
+type BaseModalProps = WindowDimensionsProps &
+    ModalProps & {
+        /** Decides whether the modal should cover fullscreen. FullScreen modal has backdrop */
+        fullscreen?: boolean;
+
+        /** Should we close modal on outside click */
+        shouldCloseOnOutsideClick?: boolean;
+
+        /** Should we announce the Modal visibility changes? */
+        shouldSetModalVisibility?: boolean;
+
+        /** Callback method fired when the user requests to close the modal */
+        onClose: () => void;
+
+        /** State that determines whether to display the modal or not */
+        isVisible: boolean;
+
+        /** Callback method fired when the user requests to submit the modal content. */
+        onSubmit?: () => void;
+
+        /** Callback method fired when the modal is hidden */
+        onModalHide?: () => void;
+
+        /** Callback method fired when the modal is shown */
+        onModalShow?: () => void;
+
+        /** Style of modal to display */
+        // type: PropTypes.oneOf(_.values(CONST.MODAL.MODAL_TYPE)),
+        type?: ValueOf<typeof CONST.MODAL.MODAL_TYPE>;
+
+        /** The anchor position of a popover modal. Has no effect on other modal types. */
+        popoverAnchorPosition?: {
```

Comment:
Extract this to a separate type

Reply:
Done!

Reply:
@JKobrynski I was talking about `popoverAnchorPosition` 🙂

Reply:
Updated 😅